### PR TITLE
Test/1843/test for number input

### PR DESCRIPTION
**Pull request by anjtka333**
## Description

Tests for the number input were made.

### Checklist

- [x] 🔽 My branch is up-to-date with "development" branch
- [x] ✅All tests passed locally
- [x] ✨My changes working with up-to-date admin and back-end part locally, like charm
- [x] 🔗 Link pull request to issue

### CODE REVIEW

File: src/components/number-input/tests/number-input.test.js [hunk 1]
```diff
@@ -0,0 +1,47 @@
...
+    const text = screen.getByDisplayValue('1');
+    expect(text).toHaveValue('1');
+    fireEvent.click(decrementBtn);
+    expect(text).toHaveValue('1');
+  });
+  test('Test input manual', () => {
+    render(<Wrapper />);
+    const decrementBtn = screen.getByTestId('decrement');
+    const text = screen.getByDisplayValue('1');
+    expect(text).toHaveValue('1');
```
Comment (afanasieva-anastasiia) [inline src/components/number-input/tests/number-input.test.js:42]: Add an empty line before expect for readability

**Review by afanasieva-anastasiia — CHANGES_REQUESTED** [review]

File: src/components/number-input/tests/number-input.test.js [hunk 2]
```diff
@@ -0,0 +1,47 @@
+import * as React from 'react';
+import { fireEvent, render, screen } from '@testing-library/react';
+import '@testing-library/jest-dom';
+import NumberInput from '../number-input';
+
+const Wrapper = () => {
```
Comment (afanasieva-anastasiia) [inline src/components/number-input/tests/number-input.test.js:6]: What is the reason for using the Wrapper component?

**Review by afanasieva-anastasiia — CHANGES_REQUESTED** [review]

File: src/components/number-input/tests/number-input.test.js [hunk 3]
```diff
@@ -0,0 +1,105 @@
...
+  test('Test textField change', () => {
+    const text = screen.getByDisplayValue('1');
+
+    expect(text).toHaveValue('1');
+    fireEvent.change(text, { target: { value: 2 } });
+    expect(text).toHaveValue('2');
+  });
+});
+
+describe('Number input block decrement woks', () => {
```
Comment (afanasieva-anastasiia) [inline src/components/number-input/tests/number-input.test.js:55]: No need to make an additional describe block if it is different only with a mocked constant number.

**Review by afanasieva-anastasiia — APPROVED** [review]

File: src/components/number-input/tests/number-input.test.js (hunk 3 above)
Comment (olga-gchak) [reply src/components/number-input/tests/number-input.test.js]: Not exact, we use "discribe" for order to highlight also different logic of component.
In this case, make sense use it

File: src/components/number-input/tests/number-input.test.js [hunk 4]
```diff
@@ -0,0 +1,82 @@
...
+        quantity={mockQuantity}
+      />
+    );
+  });
+
+  test('Test increment fn disabled', () => {
+    const incrementBtn = screen.getByTestId('increment');
+
+    fireEvent.click(incrementBtn);
+    expect(setInputValue).toHaveBeenCalledTimes(0);
```
Comment (SirAndrii) [inline src/components/number-input/tests/number-input.test.js:72]: Try to practice with another method.
I suppose, with increment 1000 you should expect toEqual(1000) and with decrementing twice - 998.

File: src/components/number-input/tests/number-input.test.js [hunk 5]
```diff
@@ -0,0 +1,87 @@
...
+  });
+
+  test('Test textField change', () => {
+    const text = screen.getByTestId('text');
+    fireEvent.change(text, { target: { value: 2 } });
+    expect(text).toHaveValue('2');
+  });
+});
+
+describe('Test decrement', () => {
```
Comment (Pipuss) [inline src/components/number-input/tests/number-input.test.js:52]: I think you need to work with describe-block naming in this file

Comment (Pipuss) [inline src/components/number-input/tests/number-input.test.js:52]: I think you need to work with describe-block naming in this file

**Review by Pipuss — DISMISSED** [review]

File: src/components/number-input/tests/number-input.test.js (hunk 5 above)
Comment (Pipuss) [reply src/components/number-input/tests/number-input.test.js:52]: And we have to name the file like "test-file.spec.js"

**Review by maarekua — APPROVED** [review]

**Review by rostyslaw27 — APPROVED** [review]

**Pull request merged**